xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -440,3 +420,315 @@ bool CEnumeratorHD::IsSDRSupported()
+  const std::vector<DXGI_FORMAT> outputFormats = GetProcessorRGBOutputFormats();
+
+  // Color spaces supported directly by the swap chain - as a set for easy lookup
+  const std::vector<DXGI_COLOR_SPACE_TYPE> bbcs =
+      DX::DeviceResources::Get()->GetSwapChainColorSpaces();
+  const std::set<DXGI_COLOR_SPACE_TYPE> backbufferColorSpaces(bbcs.begin(), bbcs.end());
+
+  // The input format cannot be worked around and is fixed.
+  // Restrieve all combinations of the allowed
+  // - input color spaces
+  // - output formats
+  // - output color spaces
+
+  const ConversionsList conversions =
+      ListConversions(inputFormat, ycbcrColorSpaces, outputFormats, rgbColorSpaces);
+
+  std::string conversionsString;
+
+  for (const Conversion& c : conversions)
+  {
+    conversionsString.append("\n");
+    conversionsString.append(StringUtils::Format(
+        "{} {} / {}{} {:<{}} to {} {:<{}} / {}{} {:<{}}", "*",
+        DX::DXGIFormatToString(c.m_inputFormat), (c.m_inputCS == heuristicsInputCS) ? "*" : " ",
+        (c.m_inputCS == inputNativeCS) ? "N" : " ", DX::DXGIColorSpaceTypeToString(c.m_inputCS), 32,
+        (c.m_outputFormat == outputFormat) ? "*" : " ", DX::DXGIFormatToString(c.m_outputFormat),
+        26, (c.m_outputCS == heuristicsOutputCS) ? "*" : " ",
+        (backbufferColorSpaces.find(c.m_outputCS) != backbufferColorSpaces.end()) ? "bb" : "  ",
+        DX::DXGIColorSpaceTypeToString(c.m_outputCS), 32));
+  }
+
+  CLog::LogFC(LOGDEBUG, LOGVIDEO,
+              "Supported conversions from format {}\n(*: values picked by "
+              "heuristics, N native input color space, bb supported as swap chain backbuffer){}",
+              DX::DXGIFormatToString(inputFormat), conversionsString);
+}
+
+bool CEnumeratorHD::IsFormatSupportedInput(DXGI_FORMAT format)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+  return IsFormatSupportedInternal(format, D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_INPUT);
+}
+bool CEnumeratorHD::IsFormatSupportedOutput(DXGI_FORMAT format)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+  return IsFormatSupportedInternal(format, D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_OUTPUT);
+}
+
+bool CEnumeratorHD::IsFormatSupportedInternal(DXGI_FORMAT format,
+                                              D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT support) const
+{
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return false;
+
+  UINT uiFlags;
+  if (S_OK == m_pEnumerator->CheckVideoProcessorFormat(format, &uiFlags))
+  {
+    if (uiFlags & support)
+      return true;
+  }
+
+  CLog::LogF(LOGERROR, "unsupported format {} for {}.", DX::DXGIFormatToString(format),
+             DX::D3D11VideoProcessorFormatSupportToString(support));
+  return false;
+}
+
+ComPtr<ID3D11VideoProcessor> CEnumeratorHD::CreateVideoProcessor(UINT RateConversionIndex)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return {};
```

Comment:
```suggestion
    return nullptr;
```
copy paste error?

Reply:
it wasn't and same thing line 689. The return type is a ComPtr not a C pointer. An unattached ComPtr is not null even if it compares as equal to a nullptr.
The compiler accepts both, I'm not sure about any possible side-effects or if one way is sub-optimal because of extra call to constructor maybe.

Reply:
If returning a POD type like void* use `nullptr`. If returning an object then {} is good.